### PR TITLE
authz: use mockgen instead of handwritten mocks for github client

### PR DESCRIPTION
**Pull request by sashaostrikov**
**🚨99% OF ADDED LINES ARE GENERATED🚨**

Closes https://github.com/sourcegraph/sourcegraph/issues/33232

## Test plan
Unit tests should pass

### CODE REVIEW

File: doc/dev/background-information/sg/reference.md [hunk 1]
```diff
@@ -87,6 +87,7 @@ Available commands in `sg.config.yaml`:
 * codeintel-worker
 * debug-env
 * docsite
+* enterprise-frontend
```
Comment (ryanslade) [inline doc/dev/background-information/sg/reference.md:90]: Did you mean to add this?

**Review by ryanslade — APPROVED** [review]
Nice

File: doc/dev/background-information/sg/reference.md (hunk 1 above)
Comment (sashaostrikov) [reply doc/dev/background-information/sg/reference.md:90]: it was generated with `sg generate` I don't really know where it came from

Comment (ryanslade) [reply doc/dev/background-information/sg/reference.md:90]: Cool, probably safe to leave it in then

Comment (sashaostrikov) [reply doc/dev/background-information/sg/reference.md]: that's should have been some local glitch on my side

**Review by mollylogue — APPROVED** [review]

**Pull request merged**